### PR TITLE
Fix runtime error on startup

### PR DESCRIPTION
**Pull request by ZirixCZ**
Fixes runtime error on startup when network's unavailable

#### Before

https://github.com/Vencord/Vesktop/assets/49836430/7c17cae7-647b-4958-86e1-d7eacd039ec5

#### After

https://github.com/Vencord/Vesktop/assets/49836430/25da02f8-ddc7-41cd-8a40-60dda520fc84

### CODE REVIEW

File: src/main/utils/http.ts [hunk 1]
```diff
@@ -20,26 +20,40 @@ export async function downloadFile(url: string, file: string, options: RequestOp
...
             if (statusCode! >= 400) return void reject(`${statusCode}: ${statusMessage} - ${url}`);
             if (statusCode! >= 300) return simpleReq(headers.location!, options).then(resolve).catch(reject);
-
             resolve(res);
+        }).on("error", err => {
+            if (retries > 10) {
+                reject(err);
+            } else {
+                setTimeout(() => resolve(simpleReq(url, options, retries + 1, backoffDelay * 2)), backoffDelay);
+            }
```
Comment (rushiiMachine) [inline src/main/utils/http.ts:40]: Shouldn't this be instead recursing backwards until 0?
The `retries` parameter would instead default to 10.

File: src/main/utils/http.ts [hunk 2]
```diff
@@ -20,26 +20,40 @@ export async function downloadFile(url: string, file: string, options: RequestOp
...
-        res.once("error", reject);
-        res.on("data", chunk => chunks.push(chunk));
-        res.once("end", () => resolve(Buffer.concat(chunks)));
-    });
+            res.once("error", reject);
+            res.on("data", chunk => chunks.push(chunk));
+            res.once("end", () => resolve(Buffer.concat(chunks)));
+        });
+    } catch (e) {
+        console.log(e);
```
Comment (rushiiMachine) [inline src/main/utils/http.ts:57]:
```suggestion
        console.error(e);
```

**Review by rushiiMachine — CHANGES_REQUESTED** [review]

File: src/main/utils/http.ts (hunk 1 above)
Comment (ZirixCZ) [reply src/main/utils/http.ts]: Refactored it https://github.com/Vencord/Vesktop/pull/432/commits/eaabc3aae9a8fc44f1887c5dc4c4a2537e190515

**Comment by Vendicated** [comment]
thanks but like i said, we should also switch to the fetch api

thus superseded by https://github.com/Vencord/Vesktop/commit/97267ef89a24f4fefae256481bf9bd8380a1d3d4

**Pull request closed without merging**